src/worker/ingestion/action-manager.ts

```diff
@@ -20,48 +22,57 @@ export class ActionManager {
         this.ready = true
     }
 
-    public getAction(id: Action['id']): Action | undefined {
+    public getTeamActions(teamId: Team['id']): ActionMap | null {
         if (!this.ready) {
             throw new Error('ActionManager is not ready! Run actionManager.prepare() before this')
         }
-        return this.actionCache[id]
+        return this.actionCache[teamId] || null
     }
 
     public async reloadAllActions(): Promise<void> {
-        this.actionCache = await this.db.fetchAllActionsMap()
+        this.actionCache = Object.fromEntries(
+            Object.entries(groupBy(await this.db.fetchAllActions(), 'team_id')).map(([teamId, actions]) => [
```

Comment:
Do you expect `db.fetchAllActions()` to be used elsewhere? If not, I think it makes things a lot simpler to just have `db.fetchAllActionsByTeam()` which returns them grouped by team?

Reply:
:+1: That's sensible

Reply:
Sorry if I was a bit unclear: I meant `fetchAllActionsGroupedByTeam` to replace `fetchAllActions` (which means getting rid of all the groupBy logic, since we're building the grouping from scratch, just like in `fetchAllActions()` right now.

Thoughts?

Reply:
Ah, I see what you mean, done. Though maybe I'll leave `groupBy` in `utils` since maybe and will come in useful and I've already written it

Reply:
Works!

Reply:
> "maybe it'll come in useful" 
... is a dangerous thing to say :D. See for example the HTTP server (let's remove it) :).

Reply:
LOL, well, at least it's isolated _and_ tested. Unlike the HTTP server… pending removal in #282 for some time